src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy

```diff
@@ -0,0 +1,169 @@
+    }
+
+    def "sets the api level onto the project settings file, then unsets it"() {
+        given: "a valid api compatibility level to set"
+        buildFile << """
+            unity {
+               setApiCompatibilityLevel("${expectedAPICompatibilityLevel.toString()}") 
+            }
+            
+        """.stripIndent()
+
+        and: "ensure that the api compatibility level isn't the default"
+        assert settings.text.contains("apiCompatibilityLevel: ${defaultAPICompatibilityLevel.value}")
+
+        when:
+        def result = runTasksSuccessfully("test")
+
+        then:
+        !result.wasSkipped("test")
+        result.wasExecuted("setAPICompatibilityLevel")
+        result.wasExecuted("unsetAPICompatibilityLevel")
+        !settings.text.contains("apiCompatibilityLevel: ${expectedAPICompatibilityLevel.value}")
+        result.standardOutput.contains("Setting API compatibility level to ${expectedAPICompatibilityLevel}")
+        result.standardOutput.contains("Setting API compatibility level to ${defaultAPICompatibilityLevel}")
+
+        where:
+        expectedAPICompatibilityLevel = APICompatibilityLevel.net4_6
+        defaultAPICompatibilityLevel = APICompatibilityLevel.net2_0_subset
+    }
+
+    def "skips if the api level is the same as the default"() {
+        given: "a valid api compatibility level to set"
+        buildFile << """
+            unity {
+               setApiCompatibilityLevel("${expectedAPICompatibilityLevel.toString()}") 
+            }
+            
+        """.stripIndent()
+
+        when:
+        def result = runTasksSuccessfully("test")
+
+        then:
+        !result.wasSkipped("test")
+        result.wasSkipped("setAPICompatibilityLevel")
+        result.wasSkipped("unsetAPICompatibilityLevel")
+        settings.text.contains("apiCompatibilityLevel: ${defaultAPICompatibilityLevel.value}")
+        !result.standardOutput.contains("Setting API compatibility level to ${expectedAPICompatibilityLevel}")
+        !result.standardOutput.contains("Setting API compatibility level to ${defaultAPICompatibilityLevel}")
+
+        where:
+        defaultAPICompatibilityLevel = APICompatibilityLevel.net2_0_subset
+        expectedAPICompatibilityLevel = defaultAPICompatibilityLevel
+    }
+
+//    def "skips if the api level given was invalid"() {
+//        given: "an invalid api compatibility level to set"
+//        buildFile << """
+//            unity {
+//               setApiCompatibilityLevel("${expectedAPICompatibilityLevel}")
+//            }
+//
+//        """.stripIndent()
+//
+//        when:
+//        def result = runTasksWithFailure("test")
+//
+//        then:
+//        // check for specific error message
+//        result.standardError.contains(SetAPICompatibilityLevel.parseFailureMessage)
+//
+//        where:
+//        defaultAPICompatibilityLevel = APICompatibilityLevel.net2_0_subset
+//        expectedAPICompatibilityLevel << [5, "lol"]
+//    }
```

Comment:
? why is this commented out?

Reply:
This was one of the earlier tests I tried to write but wasn't sure if it would reflect the intended behaviour.
Since this was more about testing how API comp level would be parsed, I tested it there (for integers, as strings and null will throw).
If an invalid type is passed for the argument it will throw since the signature won't match. Is this something we handle with overloads or is that unneeded?

Reply:
As in write a new override on DefaultUnityPLuginExtension for type Object and handle this there?

Reply:
I think if you handle the parsing in the Unittest for the enum it should be enough.

Reply:
Very well.